static_solvers/instance2data.py

```diff
@@ -1,80 +1,57 @@
 import numpy as np
 from pyvrp import Client, ProblemData, VehicleType
 
-_INT_MAX = np.iinfo(np.int32).max
+from VrpInstance import VrpInstance
 
 
-def instance2data(instance: dict) -> ProblemData:
+def instance2data(instance: VrpInstance) -> ProblemData:
     """
     Converts an instance to a ``pyvrp.ProblemData`` instance.
     """
-    dimension = instance["demands"].size
-    depots: np.ndarray = instance.get("depot", np.array([0]))
-    num_vehicles: int = instance.get("vehicles", dimension - 1)
-    capacity: int = instance.get("capacity", _INT_MAX)
-    distances: np.ndarray = instance["duration_matrix"]
-    demands: np.ndarray = instance["demands"]
-    coords: np.ndarray = instance["coords"]
-    service_times: np.ndarray = instance["service_times"]
-    durations = distances
-    time_windows: np.ndarray = instance["time_windows"]
-
-    # Default release time is zero
-    release_times = instance.get(
-        "release_times", np.zeros(dimension, dtype=int)
-    )
-
-    # Default dispatch time is planning horizon
-    horizon = instance["time_windows"][0][1]  # depot latest tw
-    dispatch_times = instance.get(
-        "dispatch_times", np.ones(dimension, dtype=int) * horizon
-    )
-
-    # Default prize is zero
-    prizes = instance.get("prizes", np.zeros(dimension, dtype=int))
-
-    # Checks
-    if len(depots) != 1 or depots[0] != 0:
+    # Check if the data is valid.
+    if not instance.is_depot[0] and instance.is_depot.sum() == 1:
         raise ValueError(
-            "Source file should contain single depot with index 1 "
-            + "(depot index should be 0 after subtracting offset 1)"
+            "First request in the instance should be the depot,",
+            "only a single depot is allowed.",
         )
 
-    if demands[0] != 0:
+    if instance.demands[0] != 0:
         raise ValueError("Demand of depot must be 0")
 
-    if service_times[0] != 0:
+    if instance.service_times[0] != 0:
         raise ValueError("Depot service duration must be 0")
 
-    if release_times[0] != 0:
+    if instance.release_times[0] != 0:
         raise ValueError("Depot release time must be 0")
 
-    if dispatch_times[0] != time_windows[0, 1]:
+    if instance.dispatch_times[0] != instance.time_windows[0, 1]:
         raise ValueError("Depot end of time window must be dispatch time")
 
-    if (time_windows[:, 0] > time_windows[:, 1]).any():
+    if (instance.time_windows[:, 0] > instance.time_windows[:, 1]).any():
         raise ValueError("Time window cannot start after end")
```

Comment:
Aren't these checks already done by `Client` or `ProblemData`?

Reply:
You're right. I removed most of them, added a TODO for the dispatch time one because I haven't built that into PyVRP yet.